src/main/java/com/arpnetworking/configuration/jackson/akka/ActorBuilderDeserializer.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2016 InscopeMetrics, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.configuration.jackson.akka;
+
+import akka.actor.Props;
+import com.arpnetworking.commons.builder.Builder;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import java.io.IOException;
+
+/**
+ * Deserializer that will create an ActorBuilder for the given actor, then create a Props from Guice.
+ *
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public class ActorBuilderDeserializer extends JsonDeserializer<Props> {
+    /**
+     * Public constructor.
+     *
+     * @param mapper the {@link ObjectMapper} to use to deserialize the {@link Builder}
+     */
+    public ActorBuilderDeserializer(final ObjectMapper mapper) {
+        _mapper = mapper;
+    }
+
+    @Override
+    public Props deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
+        final TreeNode treeNode = p.readValueAsTree();
+        final String type = ((TextNode) treeNode.get("type")).textValue();
```

Comment:
Hrm. Ok.

The use case seems to me be a combination of 1) Polymorphism; different builder types as determined by a field; 2) The use of a builder. In the first we've either done explicit type request via a TypeReference or annotation based binding via a type field. 

Props is final so you can't wrap it or extend it in order to annotate it.

http://doc.akka.io/japi/akka/2.4.0/akka/actor/Props.html

The problem seems to be that we're storing any Props here for the _clusterJoinerActor which is not type-complete (I'm guess not just any actor can be a "joiner actor").

For example, if the Props were wrapped with NonJoiningClusterJoiner and we used that type here (or more correctly an interface of that type) then we would be just to a manageable polymorphic deserialization problem (e.g. annotate the subclasses of that interface). Using Props directly corners us in many ways, and doesn't help describe what is actually required here.

Happy to discuss.